File: src/main/java/org/profit/server/handler/action/Action.java
package org.profit.server.handler.action;

import io.netty.channel.ChannelHandlerContext;
import org.profit.server.handler.communication.RequestMessage;
import org.profit.server.handler.communication.exception.InvalidParamException;

/**
 * Created by TangYing
 */
public interface Action {

    /**
     * Interface of Action
     */
    public void execute(ChannelHandlerContext ctx, RequestMessage request) throws InvalidParamException;
}

File: src/main/java/org/profit/server/handler/action/Action.kt
package org.profit.server.handler.action

import io.netty.channel.ChannelHandlerContext
import org.profit.server.handler.communication.RequestMessage
import org.profit.server.handler.communication.exception.InvalidParamException

/**
 * Created by TangYing
 */
interface Action {
    /**
     * Interface of Action
     */
    @Throws(InvalidParamException::class)
    fun execute(ctx: ChannelHandlerContext?, request: RequestMessage)
}